File: src/py_pubsub/py_pubsub/subscriber_node.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class PySubscriber(Node):
    def __init__(self):
        super().__init__('py_subscriber')
        self.subscription = self.create_subscription(
            String, 'py_topic', self.listener_callback, 10)

    def listener_callback(self, msg):
        self.get_logger().info(f'Python Sub received: "{msg.data}"')

def main(args=None):
    rclpy.init(args=args)
    node = PySubscriber()
    rclpy.spin(node)
    node.destroy_node()
    rclpy.shutdown()

File: Tasks/Task_1/src/cpp_pubsub/src/subscriber_node.cpp
#include "rclcpp/rclcpp.hpp"
#include "std_msgs/msg/string.hpp"

class CppSubscriber : public rclcpp::Node {
public:
  CppSubscriber() : Node("cpp_subscriber") {
    subscription_ = create_subscription<std_msgs::msg::String>(
      "cpp_topic", 10,
      [this](const std_msgs::msg::String& msg) {
        RCLCPP_INFO(get_logger(), "C++ Sub received: '%s'", msg.data.c_str());
      });
  }
private:
  rclcpp::Subscription<std_msgs::msg::String>::SharedPtr subscription_;
};

int main(int argc, char* argv[]) {
  rclcpp::init(argc, argv);
  rclcpp::spin(std::make_shared<CppSubscriber>());
  rclcpp::shutdown();
  return 0;
}
